pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -282,12 +282,17 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 
 		if isExposed(infinispan) {
 			externalService := r.serviceExternal(infinispan)
-			err = r.client.Create(context.TODO(), externalService)
-			if err != nil && !errors.IsAlreadyExists(err) {
-				reqLogger.Error(err, "failed to create external Service", "Service", ser)
-				return reconcile.Result{}, err
+			if externalService != nil {
+				err = r.client.Create(context.TODO(), externalService)
```

Comment:
@rigazilla what do you think to return back NodePort to the Infinispan structure after create? I must be more visible for users.

Reply:
Maybe we could add to the infinispan RC a reference to the created service. wdyt @dmvolod ?

Reply:
but yeah, also update the struct with the actual values is a good idea

Reply:
I would go for your solution @dmvolod, we already provide selector to join service to Infinispan